src/benchq/resource_estimation/decoder_resource_estimator.py

```diff
@@ -0,0 +1,96 @@
+import warnings
+from typing import Optional
+
+from benchq.data_structures.decoder import DecoderModel
+from benchq.data_structures.hardware_architecture_models import BasicArchitectureModel
+from benchq.data_structures.resource_info import DecoderInfo
+
+
+def get_decoder_info(
+    hw_model,
+    decoder_model: Optional[DecoderModel],
+    code_distance: int,
+    space_time_volume: float,
+    n_logical_qubits: int,
+):
+    if not decoder_model:
+        return None
+    else:
+        speed_limit = get_decoder_distance_limit_due_to_speed(hw_model, decoder_model)
+        if code_distance >= decoder_model.highest_calculated_distance:
+            warnings.warn(
+                f"Code distance {code_distance} is too high to get resource estimates "
+                "because resource estimates have not been calculated for "
+                f"distances beyond {decoder_model.highest_calculated_distance}.",
+                RuntimeWarning,
+            )
+            return None
+        elif code_distance > speed_limit:
+            warnings.warn(
+                f"Code distance {code_distance} is too high to be decoded "
+                "because the decoder is too slow.",
+                RuntimeWarning,
+            )
+            return None
+        else:
+            decoder_total_energy_consumption = (
+                space_time_volume
+                * decoder_model.power_in_nanowatts(code_distance)
+                * decoder_model.delay_in_nanoseconds(code_distance)
+            )
```

Comment:
I think there's a factor of 1e9 missing here since we want the energy in nJ, but 1 nW x 1 ns = 1e-9 nJ.

Reply:
Yup! Thanks for catching that!